src/SelectInput.js

```diff
@@ -98,21 +100,31 @@ class SelectInput extends PureComponent {
 			const lastIndex = (hasLimit ? limit : items.length) - 1;
 			const atFirstIndex = selectedIndex === 0;
 			const nextIndex = (hasLimit ? selectedIndex : lastIndex);
+			const nextRotateIndex = atFirstIndex ? rotateIndex + 1 : rotateIndex;
+			const nextSelectedIndex = atFirstIndex ? nextIndex : selectedIndex - 1;
 
 			this.setState({
-				rotateIndex: atFirstIndex ? rotateIndex + 1 : rotateIndex,
-				selectedIndex: atFirstIndex ? nextIndex : selectedIndex - 1
+				rotateIndex: nextRotateIndex,
+				selectedIndex: nextSelectedIndex
 			});
+
+			const slicedItems = hasLimit ? arrRotate(items, nextRotateIndex).slice(0, limit) : items;
+			onHighlight(slicedItems[nextSelectedIndex]);
```

Comment:
Could it be called with the `index` as well? This is the UI I want to replicate using this module:

![image](https://user-images.githubusercontent.com/1404810/54865511-af073980-4d66-11e9-9672-04367592416b.png)

Reply:
Which index, the index of the hightlighted item in the entire items array or the index in the slicedItems?

Reply:
the index in the entire array was what I was thinking about, but maybe both?

Reply:
Looking at the code again, it seems as though we only have the index related to slicedItems when a limit is set and the index related to all items when no limit is set.

Thinking this might be better as another PR adding the desired functionality to both onSelect and onHighlight as there might have to incl some refactoring.

Reply:
Good suggestion, let's do it separately and pass `index` as a second argument to `onSelect` and `onHighlight` like @zimme suggested.